packages/itmat-types/src/types/file.ts

```diff
@@ -1,14 +1,53 @@
-export interface IFile {
-    id: string;
+import { Readable } from 'stream';
+import { IBase } from './base';
+
+export interface IFile extends IBase {
+    studyId: string | null; // null for system and user file
+    userId: string | null; // null for system and study file
     fileName: string;
-    studyId: string;
-    projectId?: string;
-    fileSize?: string;
-    description: string;
-    uploadTime: string;
-    uploadedBy: string;  // userId
+    fileSize: number;
+    description?: string;
+    properties: Record<string, unknown>;
     uri: string;
-    deleted: number | null;
     hash: string;
-    metadata: Record<string, unknown>
+    fileType: enumFileTypes;
+    fileCategory: enumFileCategories;
+    sharedUsers?: string[];
 }
+
+
+
+export enum enumFileTypes {
```

Comment:
What is the motivation behind maintaining a record for the file type ? I understand this only here to represent the extension of the file ? I'm not sure it is useful ?

Reply:
For thie PR it is more like a placeholder. 
I use that for validating the file type if needed (e.g., v3 allows upload profile (images) so only JPG, PNG, etc could be uploaded); and in further if there is a need to parse the file then we can design parsing methods based on the file extension.